lms/services/user.py

```diff
@@ -145,10 +146,10 @@ def get_users(  # noqa: PLR0913, PLR0917
 
         if instructor_h_userid:
             instructor_h_userid_clause = User.id.in_(
-                select(AssignmentMembership.user_id)
-                .join(Assignment)
-                .where(
-                    Assignment.course_id.in_(
+                # GroupingMembership has membership information, but it lacks role info
```

Comment:
So to recap, we use the query in ` CourseService.course_ids_with_role_query` to get all the course IDs where the user is an instructor.

And, instead of joining AssignmentMembership -> Assignment table able to filter on Assignment.course_id we filtering directly on GroupingMembership, which does relate users to courses.

We avoid one join for some performance speedup and a bit (very little) sanity to the whole query.

As with similar queries, the right solution is to have a proper GroupingMembership table where we can query this in one go, out of scope here.

Reply:
> How does the grouping_membership table get populated? Do we add entries to it for the course at the same time we add assignment-user relations?

Yes, both get populated/updated at the same time, during the launch.

> Can you expand on this? What are you suggesting the improved schema would look like?

If we had roles on GroupingMembership we wouldn't need to go 
AssingmentMmebership -> AssignmentGrouping -> GroupingMembership 

We could do the role filtering directly on one table.